src/vue/modules/coinpayments-deposit/components/timeout-ticker.vue

```diff
@@ -0,0 +1,62 @@
+<template>
+  <div class="timeout-ticker">
+    <template v-if="secondsLeft > 0">
+      {{ 'coinpayments-deposit.time-left-msg' | globalize }}:
+      {{ timeLeftFormatted }}
+    </template>
+    <template v-else>
+      {{ 'coinpayments-deposit.time-over-msg' | globalize }}
+    </template>
+  </div>
+</template>
+
+<script>
+import moment from 'moment'
+
+export default {
+  name: 'timeout-ticker',
+  props: {
+    endTime: { type: Number, required: true }, // unix timeStamp
+  },
+  data: _ => ({
+    secondsLeft: null,
+    intervalId: null,
+    moment,
+  }),
+  computed: {
+    timeLeftFormatted () {
+      const timeLeft = moment
+        .duration(this.secondsLeft, 'seconds')
+        .asMilliseconds()
+
+      if (timeLeft <= 0) {
+        return '00:00:00'
+      }
+
+      return moment
+        .utc(timeLeft)
+        .format('HH:mm:ss')
+    },
+  },
+  created () {
+    this.secondsLeft = (this.endTime - moment().unix())
+    this.startTicker()
+  },
+  beforeDestroy () {
+    clearInterval(this.intervalId)
+  },
+  methods: {
+    startTicker () {
+      this.intervalId = setInterval(() => {
+        this.secondsLeft--
```

Comment:
on timers on JS you should never do `seconds--`. instead you should every second check the timestamp and calc the time left

Reply:
ok